command/e2etest/providers_mirror_test.go

```diff
@@ -0,0 +1,76 @@
+package e2etest
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sort"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/hashicorp/terraform/e2e"
+)
+
+// The tests in this file are for the "terraform providers schema" command,
+// which is tested in an e2etest mode rather than a unit test mode because it
+// interacts directly with Terraform Registry and the full details of that are
+// tricky to mock. Such a mock is _possible_, but we're using e2etest as a
```

Comment:
I am not at all opposed to this e2e test, but just so you know we recently added a `testRegistrySource` helper function in the command package:

https://github.com/hashicorp/terraform/blob/020084f6d044702d3a27dbc16800782e2bedb50e/command/command_test.go#L919

Reply:
This particular one is expecting to be talking always to upstream registries (so you don't have to unconfigure your mirrors just to populate them), so it wasn't very amenable to that sort of mocking. I did have an earlier version that created a mock registry on a fake domain and forced use of it via service discovery, but that required the fake providers to be properly signed and that was a little too complex to set up in a reasonable amount of time to get this done right now. Maybe in a later commit!